packages/product/src/services/product-module-service.ts

```diff
@@ -558,6 +588,11 @@ export default class ProductModuleService<
       sharedContext
     )
 
+    await this.eventBusService_?.emit<ProductCategoryServiceTypes.ProductCategoryEventData>(
```

Comment:
todo: I just realised, the write method of the module service should all be bulk, can I let you to update that in a separate pr. This method for example should allow to create multiple categories

Reply:
As mentioned over huddle, tree repositories have to stay singular due to the potential conflicting events messing with ranking in the same transaction.